app/controllers/devise_token_auth/concerns/set_user_by_token.rb

```diff
@@ -125,6 +125,19 @@ def resource_class(m=nil)
     mapping.to
   end
 
+  def get_resource(uid_field='uid', uid=nil)
+    if DeviseTokenAuth.custom_scope
+      resource_class.send(DeviseTokenAuth.custom_scope, uid).first
+    else
+      q = "#{uid_field.to_s} = ? AND provider='email'"
+
+      if ActiveRecord::Base.connection.adapter_name.downcase.starts_with? 'mysql'
+        q = "BINARY " + q
```

Comment:
I wasn't trying to deviate from the original authors code too much to lessen the diffs but whatever merge would probably choke on this anyway.